From e840999ffc93ecb2d870dfed334e093ca1e5e65b Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Fri, 27 Nov 2009 06:20:13 +0000
Subject: dix: remove core devices when shutting down. (#25028)

NewInputDeviceRequest (and RemoveDevice) have checks in place to not allow
removal of the VCP/VCK. When shutting down, they need to be cleaned up
nonetheless to free the memory associated.

X.Org Bug 25028 <http://bugs.freedesktop.org/show_bug.cgi?id=25028>

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
Reviewed-by: Keith Packard <keithp@keithp.com>
(cherry picked from commit 66bb8c6fbdfc0fc0d971aac4ec6f949bb9288c1b)
---
diff --git a/dix/devices.c b/dix/devices.c
index 395e19a..e3fd456 100644
--- a/dix/devices.c
+++ b/dix/devices.c
@@ -908,6 +908,9 @@ CloseDownDevices(void)
         DeleteInputDeviceRequest(dev);
     }
 
+    CloseDevice(inputInfo.pointer);
+    CloseDevice(inputInfo.keyboard);
+
     inputInfo.devices = NULL;
     inputInfo.off_devices = NULL;
     inputInfo.keyboard = NULL;
--
cgit v0.8.3-6-g21f6
